### DISCUSSION TITLE
Multi-game mode and multi-difficulty lobbies

### DISCUSSION THREAD

**Discussion by Nekomosh004** [ppy/osu#16152] [Ideas]
I'm unsure if this is something that has already been posted here, or in the [osu! Feature Request sub forum](https://osu.ppy.sh/community/forums/4) _(likely)_, but still wanted to post here just in case it is not here.

**The idea:**
To enable the lobby Host to select one beatmap, then enable all joining players to select which Game Mode (or ruleset), and/or Difficulty from that one beatmap. Its like Free Mod for mods, but for gamemode and beatmap difficulty.

I have friends in osu! that plays well on standard, but I don't, and that somehow frustrates me that I cannot play with them because _I suck_ in osu! standard, but decent with osu! catch. This feature, for sure, will bridge all players across all gamemodes to play any difficulty on one multiplayer room, and I think thats exciting >~<

**_What do ya'll think?_** Hopefully this is something that can be implemented with the next major release. :>

**Comment by bdach** [comment]
For being able to choose between difficulties, see https://github.com/ppy/osu/issues/5578

For ruleset mixing, I believe in playlists that is already supported, and I *think* it should also work in multiplayer but I haven't verified this manually.

**Reply by peppy** [reply]
Yep, that request should cover different rulesets too.
